library/core/src/main/java/com/google/android/exoplayer2/scheduler/Requirements.java

```diff
@@ -162,6 +171,34 @@ private boolean isDeviceIdle(Context context) {
         : Util.SDK_INT >= 20 ? !powerManager.isInteractive() : !powerManager.isScreenOn();
   }
 
+  /**
+   * Implementation taken from the the WorkManager source.
+   * @see <a href="https://android.googlesource.com/platform/frameworks/support/+/androidx-master-dev/work/workmanager/src/main/java/androidx/work/impl/constraints/trackers/StorageNotLowTracker.java">StorageNotLowTracker</a>
+   */
+  private boolean isStorageNotLow(Context context) {
+    IntentFilter intentFilter = new IntentFilter();
+    intentFilter.addAction(Intent.ACTION_DEVICE_STORAGE_OK);
+    intentFilter.addAction(Intent.ACTION_DEVICE_STORAGE_LOW);
```

Comment:
@jdegroot-dss It works only for internal storage. When downloading to external storage, this method of determining is useless and even false. If there is no space in the internal storage, we cannot download to the external storage when this requirement is enabled

Reply:
@jdegroot-dss Does this affect your intended usage at all, or is it fine for your use case? It still seems OK to support this to me, although perhaps this should be made clearer in the documentation. If someone is placing content onto external storage they can just not use the feature.

If it's still useful for you, please just confirm this here and I'll look at getting it merged in. Thanks!

Reply:
Thanks for the heads up @vigneshvg, I was unaware of that limitation. 

@ojw28 I don't think this will affect our use case and it would still be useful to have. Agreed with updating the documentation. Do we want to restrict the usage of the requirement if downloads are going to external storage?

Reply:
I think just making it clear in the documentation is enough.